src/openrct2/ride/water/SplashBoats.cpp

```diff
@@ -1276,7 +1276,7 @@ void vehicle_visual_splash_boats_or_water_coaster(
     }
 
     session.CurrentlyDrawnEntity = vehicleToPaint;
-    imageDirection = ((session.CurrentRotation * 8) + vehicleToPaint->sprite_direction) & 0x1F;
+    imageDirection = OpenRCT2::Entity::Yaw::MaskYaw(OpenRCT2::Entity::Yaw::YawFrom4(session.CurrentRotation) + vehicleToPaint->sprite_direction);
```

Comment:
This doesn't look too nice perhaps an `Yaw::Add` would be nicer:
```
imageDirection = OpenRCT2::Entity::Yaw::Add(vehicleToPaint->sprite_direction, OpenRCT2::Entity::Yaw::YawFrom4(session.CurrentRotation));
```
I feel that is easier to understand than talking about masks.